data/modules/Scout/ScanDisplay.lua

```diff
@@ -191,27 +193,41 @@ function scanDisplay:drawScanInfo(scan, isHighlighted)
 
 	if scan.orbital then
 		target = string.format("%.1f%%", scan.targetCoverage * 100.0)
+		completion = string.format("%2.1f%%", math.floor(scan.coverage * 1000) / 10)
 	else
 		target = ui.Format.Distance(scan.targetCoverage * 1000.0, "%.1f")
+		completion = ui.Format.Distance(scan.coverage * 1000.0, "%.1f")
 	end
 
-	local completion = math.min(1.0, scan.coverage / scan.targetCoverage)
-
 	local data = {
 		title = sBody.name .. ", " .. scan.bodyPath:GetStarSystem().name,
 		target = target,
-		completion = string.format("%2.1f%%", completion * 100.0),
+		completion = completion,
 		isActive = self.scanMgr:GetActiveScan() == scan,
 		scan = scan,
 		icon = scan.orbital and icons.map or icons.scanner,
 		{ icons.comms, target, ls.SCAN_TARGET_COVERAGE },
 		{ icons.scanner, ui.Format.Distance(scan.minResolution, "%.1f"), ls.SCAN_MAXIMUM_SPATIAL_RESOLUTION },
 		{ icons.altitude, altitude, ls.SCAN_MAXIMUM_ALTITUDE },
+		progress = (scan.targetCoverage / scan.coverage) * 100
 	}
 
 	return ScanCard:draw(data, isHighlighted)
 end
 
+---@param scan ScanData
+function scanDisplay:drawScanProgress(scan)
+	local completion = math.min(1.0, scan.coverage / scan.targetCoverage)
+	local width = ui.getContentRegion().x
+	-- The style progress bar colour is yellow which is very jarring for this
+	-- display. So instead lets use the slider grab colour which should remain
+	-- suitable across any style changes in the future.
+	local progressBarColor = ui.getStyleColor("SliderGrabActive")
```

Comment:
Note: this should use `colors.SliderGrabActive` or (for contrast, given this is displaying a white label overtop) `colors.uiPrimary` or `colors.uiPrimaryLight`. `ui.getStyleColor()` is an old hack from before we had a consistent UI style, and explicitly referencing the style color allows easier/better refactoring.

Reply:
Fixed in 0174ee9f587352af9af65266a7abdd1372db0d1d

I chose `colors.uiPrimaryLight` just to give it a tiny bit of difference from the standard background. I do prefer the even lighter `SliderGrabActive` but it does reduce contrast, and ImGui currently does not have support for rendering progress bar text with inverted colors when being covered by the progress.

Reply:
Incidentally, there's a new version of ImGui which fixes the alignment of the text in the progress bar.

The current version of ImGui places the text into the non-progress portion of the bar which makes the text move as the progress bar fills, which looks a bit weird. The new version has a fixed place for the text inside the progress bar.

What's the procedure for updating ImGui? (I'm guessing a fair bit of testing would be required..)

Reply:
Hah... funny you ask. I've had "Update ImGui" on my TODO list for the last ~week. The short and snarky form of the procedure is "get @Web-eWorks to do it".

The actual procedure is a bit longer than that - we have several patches applied onto the base ImGui codebase, which we have stored in-tree as `contrib/imgui/patches/*.patch`. Those have to be evaluated, applied, and potentially rewritten. Then, after that's done, a pass on API breaking changes from the ImGui side has to be done, and the Lua API checked to see if it uses any of the now-deprecated functions from the ImGui API.

If Lua does use a now-removed API, all callsites have to be migrated to either a drop-in wrapper in `data/pigui/libs/wrappers.lua` or the replacement API.

Most of the testing required from an ImGui upgrade is handled by "does this compile" - ocornut is very good about documenting breaking changes and keeping APIs stable. The fun bit is exposing all of the new ImGui API functions to Lua in a way that makes sense - that is typically an iterative process that lasts until (and often past) the next ImGui upgrade.

Reply:
Oof; a bit more involved than I had anticipated.. them's the breaks when you maintain your own patches!